packages/server/graphql/mutations/endSprintPoker.ts

```diff
@@ -52,7 +53,7 @@ export default {
 
     // RESOLUTION
     // remove hovering data from redis
-    const estimatePhase = phases.find((phase) => phase.phaseType === 'ESTIMATE')!
+    const estimatePhase = phases.find((phase) => phase.phaseType === 'ESTIMATE')! as EstimatePhase
```

Comment:
It looks like we always need to cast the phase types even when filtering by `phaseType`, (see [example](https://github.com/ParabolInc/parabol/blob/027bb8d7f51d4b958af29569ec181f86d347bb5b/packages/server/graphql/mutations/updateNewCheckInQuestion.ts#L55)). 
I tried adding `!phaseType` to each db phase type such as the `EstimatePhase` (as we do in [CheckInPhase](https://github.com/ParabolInc/parabol/blob/db45e3b5a4c27e917dcc267d33175bae8a04970b/packages/server/database/types/CheckInPhase.ts#L15)) but TypeScript still reads it as the `GenericMeetingPhase`. 

In meeting types we added the `meetingType` field (see [MeetingPoker](https://github.com/ParabolInc/parabol/blob/6b83c0a272bfa0e4cad607ed3b20970b3f2a4784/packages/server/database/types/MeetingPoker.ts#L19)) and TypeScript can successfully discriminate. The only difference between meeting types and meeting phases that I see if in the meeting types, we include `meetingType` in `super`. 

I tried adding the `phaseType` in `super` in EstimatePhase but get the error `Type '{ phaseType: string; }' is not assignable to type '"vote"'`. Any ideas on how I can discriminate phase types without casting?

Reply:
@nickoferrall the only way I can think of is doing this:
```
    const estimatePhase = phases.find(
      (phase: GenericMeetingPhase): phase is EstimatePhase => phase.phaseType === 'ESTIMATE'
    )!
```
It's called type predicate, more about it here: https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates

Reply:
Yep, that's a good option too but I was hoping I could just use the `phaseType` like we do with `meetingType` [here](https://github.com/ParabolInc/parabol/blob/af9366fc995ec0cae4ac5017bf28358ba4f3554c/packages/server/graphql/mutations/helpers/notifySlack.ts#L124). By checking the `meetingType`, TypeScript knows which `meeting` type it is without using type predicates or casting but I think this is only possible with if/else or switch statments

Reply:
the built-in find method isn't too helpful here. 

might be worth it to make a helper function that does a single cast so everything that calls it doesn't have to do a cast

```ts
    // if we have a lookup table somewhere you can use that instead
    interface PhaseTypeLookup {
      ESTIMATE: EstimatePhase
      updates: UpdatesPhase
    }
    const getPhase = <T extends keyof PhaseTypeLookup>(phases: GenericMeetingPhase[], phaseType: T) => {

      return phases.find((phase) => phase.phaseType === phaseType) as unknown as PhaseTypeLookup[T]
    }


    const estimatePhase = getPhase(phases, 'ESTIMATE')
```

Reply:
I like it! I've refactored the phase casting to use this new `getPhase` function